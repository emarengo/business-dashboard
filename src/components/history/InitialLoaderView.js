import React from 'react';

import styled from '@emotion/styled';
import { css } from '@emotion/core';

const InitialLoaderContainer = styled.div`
  width: 100%;
  height: 100vh;
  position: absolute;
  left: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #093548;
  font-family: 'Jost', sans-serif;
  z-index: 7;
  overflow: hidden;
  background: #dde6f2;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    ${css`
      background: linear-gradient(90deg, #dde6f2, #eaf0f9, #dde6f2);
      animation: progress 2s ease-in-out infinite;
      @keyframes progress {
        0% {
          transform: translate3d(-100%, 0, 0);
        }
        100% {
          transform: translate3d(100%, 0, 0);
        }
      }
    `}
  }
`;

const LoaderText = styled.p`
  position: absolute;
  width: 412px;
  font-size: 17px;
  line-height: 1.31;
  font-weight: 400;
  text-align: center;
`;

const InitialLoaderView = ({ text }) => {
  return (
    <InitialLoaderContainer>
      <LoaderText data-testid="LoaderText">{text}</LoaderText>
    </InitialLoaderContainer>
  );
};

export default InitialLoaderView;
