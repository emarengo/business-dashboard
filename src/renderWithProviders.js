import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'emotion-theming';
import { BeatTheme } from '@thebeatapp/beat-ui';

const customRender = (node, ...options) => {
  return render(
    <Router>
      <ThemeProvider theme={BeatTheme}>{node}</ThemeProvider>
    </Router>,
    ...options
  );
};

export * from '@testing-library/react';
export { customRender as render };
export { customRender as rerender };
