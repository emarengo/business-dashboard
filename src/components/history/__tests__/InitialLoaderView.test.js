import React from 'react';
import { cleanup, render } from '../../../renderWithProviders';
import InitialLoaderView from '../InitialLoaderView';

describe('Initial Loader View', () => {
  let renderResult;
  let node;
  const text = 'Please Wait...';

  beforeEach(() => {
    renderResult = render(<InitialLoaderView text={text} />);
    node = renderResult.getByTestId('LoaderText');
  });

  afterEach(cleanup);

  it('should display the text prop', () => {
    expect(node).toHaveTextContent(text);
  });

  it('should match snapshot', () => {
    expect(node).toMatchSnapshot();
  });
});
