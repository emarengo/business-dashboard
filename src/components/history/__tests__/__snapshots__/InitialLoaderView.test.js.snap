// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial Loader View should match snapshot 1`] = `
<p
  class="css-13swf06"
  data-testid="LoaderText"
>
  Please Wait...
</p>
`;
